### DISCUSSION TITLE
Trigger snippet completion only in commented lines?

### DISCUSSION THREAD

**Discussion by rodhash** [L3MON4D3/LuaSnip#1276] [General]
For instance this custom snippet to ignore Typescript errors which we need to add in a commented line:
```json
  "tsignore": {
    "prefix": "@ts",
    "body": [
      "@ts-ignore"
    ],
    "description": "Suppresses @ts-check errors on the next line",
    "scope": "typescript"
  },
```

This is triggering also when I'm typing an user email address when I hit `@`.

Any way to trigger this only on a line commented out?

UPDATE: increased `keyword_length` and now it's not triggering with just one char (which was annoying). But I'm wondering if forcing a snippet to trigger only in a particular situation (comment) is a thing. I saw a vid from TJ but it doesn't seem to be the case.

**Comment by L3MON4D3** [comment]
Oh yeah, it is a thing, check `condition`.
But you'll have to define your snippet in lua for this to work, it is not available for vscode-like snippet sources.

(the general blueprint is to find the current cursor position in `condition`, then check if the cursor is inside a comment by simple parsing or by querying treesitter)
